### PR TITLE
Refactor painter and gpu

### PR DESCRIPTION
**Pull request by M-Adoo**
This pr contains the following changes: 

- The output of the `painter` package only contains fill-path and brush
- The `gpu` package implements a GPU-related painter backend abstract implementation and generates triangle drawing commands. And provides a specific implementation of `wgpu, it allowing to be turned off by a feature and provides your own specific GPU implementation. 
- The core package no longer depends on `gpu` package, and no longer depends on `winit` (not finished the event is not separated), the core window interacts with the platform window through the trait `ShellWindow`, and submits drawing commands to it. 
- `gpu` package completes the overall framework refactoring and reduces GPU overhead, but there is still a lot of room for improvement

### CODE REVIEW

File: themes/material/src/state_layer.rs [hunk 1]
```diff
@@ -43,22 +43,20 @@ impl ComposeChild for InteractiveLayer {
       Stack {
         fit: StackFit::Passthrough,
         DynWidget { id: host, dyns: child }
-        IgnorePointer {
```
Comment (sologeek) [inline themes/material/src/state_layer.rs:46]: Why don't need `IgnorePointer` here?

File: painter/src/painter.rs [hunk 2]
```diff
@@ -17,61 +16,81 @@ use std::ops::{Deref, DerefMut};
...
+    color: Color,
+  },
+  ImgPath {
+    path: PaintPath,
+    img: ShareResource<PixelImage>,
+    opacity: f32,
+  },
+  // Todo: keep rectangle clip.
+  Clip(PaintPath),
   PopClip,
```
Comment (sologeek) [inline painter/src/painter.rs:94]:
```suggestion
  PopClip,
```
here need more comment to describe it, I looked at the code for a while to know the meaning

File: gpu/src/wgpu_impl/shaders/alpha_triangles.wgsl [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+struct VertexOutput {
+  @builtin(position) pos: vec4<f32>,
+  @location(0) value: f32,
+}
+
+
+@vertex
+fn vs_main(input: VertexInput) -> VertexOutput {
+    let pos = input.pos * vec2(2., -2.) + vec2(-1., 1.);
```
Comment (sologeek) [inline gpu/src/wgpu_impl/shaders/alpha_triangles.wgsl:15]: What meaning of these numbers? 😂

File: themes/material/src/state_layer.rs (hunk 1 above)
Comment (M-Adoo) [reply themes/material/src/state_layer.rs:46]: seems my fault

File: gpu/src/wgpu_impl/shaders/alpha_triangles.wgsl (hunk 3 above)
Comment (M-Adoo) [reply gpu/src/wgpu_impl/shaders/alpha_triangles.wgsl:15]: // convert from gpu-backend coords(0..1) to wgpu corrds(-1..1)

**Review by wjian23 — APPROVED** [review]

**Pull request merged**